File: android/core/domain/src/test/kotlin/com/akaar/core/domain/MoneyTest.kt
package com.akaar.core.domain

import com.akaar.core.domain.model.Money
import org.junit.Assert.assertEquals
import org.junit.Test

class MoneyTest {

    @Test fun `formats with indian digit grouping`() {
        assertEquals("₹450", Money(45_000).format())
        assertEquals("₹1,200", Money(120_000).format())
        assertEquals("₹1,23,456", Money(12_345_600).format())
        assertEquals("₹12,34,567", Money(123_456_700).format())
    }

    @Test fun `keeps paise when present`() {
        assertEquals("₹681.45", Money(68_145).format())
        assertEquals("₹0.05", Money(5).format())
    }

    @Test fun `arithmetic stays exact`() {
        // The floor guarantee cannot rest on binary rounding, so this is integer maths.
        val unit = Money(68_145)
        assertEquals(Money(204_435), unit * 3)
        assertEquals(Money(68_145 * 7), unit * 7)
    }

    @Test fun `compares by paise`() {
        assert(Money(68_145) > Money(60_000))
        assert(Money(60_000) < Money(68_145))
    }
}

File: mobile/packages/core-domain/src/money.test.ts
// Port of android/core/domain/.../MoneyTest.kt, same cases, node:test instead
// of JUnit. Run with `pnpm --filter @akaar/core-domain test`.
import assert from "node:assert/strict";
import { test } from "node:test";
import { Money } from "./money.ts";

test("formats with indian digit grouping", () => {
  assert.equal(new Money(45_000).format(), "₹450");
  assert.equal(new Money(120_000).format(), "₹1,200");
  assert.equal(new Money(12_345_600).format(), "₹1,23,456");
  assert.equal(new Money(123_456_700).format(), "₹12,34,567");
});

test("keeps paise when present", () => {
  assert.equal(new Money(68_145).format(), "₹681.45");
  assert.equal(new Money(5).format(), "₹0.05");
});

test("arithmetic stays exact", () => {
  // The floor guarantee cannot rest on binary rounding, so this is integer maths.
  const unit = new Money(68_145);
  assert.equal(unit.times(3).paise, new Money(204_435).paise);
  assert.equal(unit.times(7).paise, new Money(68_145 * 7).paise);
});

test("compares by paise", () => {
  assert.ok(new Money(68_145).greaterThan(new Money(60_000)));
  assert.ok(new Money(60_000).lessThan(new Money(68_145)));
});
